### ISSUE TITLE
[BUG] RPM packages depend on (now deprecated) python 3.6 

### ISSUE THREAD

**Issue by synedraIT** [saltstack/salt#61475]
**Description**
Hello!
Our auditing tool detected a deprecated version of python3 (python 3.6) installed on all our Rocky/Alma Linux minions.

The last version of python 3.6 available in AlmaLinux is python 3.6.8, which, according to [this](https://www.python.org/downloads/release/python-368/) page, is end of life and should not be used anymore.
If I understood this correctly, this is a packaging issue.

**Setup**
Fully up to date AlmaLinux VM, with `python36` module disabled (see below).
  
**Steps to Reproduce the behavior**
* Install the salt-repo as in the [Download page](https://repo.saltproject.io/#rhel)
* Run `dnf remove python36` to remove python 3.6 - salt will be removed as well, as it's dependencies are not fulfilled anymore.
* Run `dnf module disable python36`
* Try to install salt again, which leads to this error:
```
dnf install salt-minion
Failed to set locale, defaulting to C.UTF-8
Last metadata expiration check: 0:01:27 ago on Tue Jan 18 09:38:16 2022.
Error:
 Problem: package salt-minion-3004-1.el8.noarch requires /usr/bin/python3.6, but none of the providers can be installed
  - conflicting requests
  - package python36-3.6.8-38.module_el8.5.0+2569+5c5719bc.x86_64 is filtered out by modular filtering
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

**Expected behavior**
Package should install, with python 3.8 as dependency.

**Comment by krionbsd** [comment]
@synedraIT please note we provide rhel8 packages with 3.6 support as a default python version

**Comment by synedraIT** [comment]
> @synedraIT please note we provide rhel8 packages with 3.6 support as a default python version

Hello! Thanks for the reply.
Python 3.6 is no longer maintained, as per https://www.python.org/downloads/release/python-368/ and https://www.python.org/downloads/release/python-3615/
>Note: The release you are looking at is Python 3.6.8, the final bugfix release for the legacy 3.6 series which has now reached end-of-life and is no longer supported. See the downloads page for currently supported versions of Python. The final source-only security fix release for 3.6 was 3.6.15.

>Note: The release you are looking at is Python 3.6.15, the final security bugfix release for the legacy 3.6 series which has now reached end-of-life and is no longer supported. See the downloads page for currently supported versions of Python. The final bugfix release for 3.6 was 3.6.8.

Will this not become a security issue in the future?

**Comment by krionbsd** [comment]
@synedraIT at the time point of 3004 release 3.6 was still supported, we're going to release new salt packages with 3.7+ as a dependency anyway

**Comment by synedraIT** [comment]
> @synedraIT at the time point of 3004 release 3.6 was still supported, we're going to release new salt packages with 3.7+ as a dependency anyway

Okay, so the problem will go away once we get a new salt release - thats great!
I personally had some trouble finding repos for 3.7 on AlmaLinux, so 3.8 might be the easier to do. I'll leave it up to you, of course!
Thank you very much for the support, if new packages are released with 3.7+ anyway, I'll just close this issue.
